src/develop/imageop.c

```diff
@@ -868,6 +868,108 @@ static void dt_iop_gui_duplicate_callback(GtkButton *button, gpointer user_data)
   dt_iop_gui_duplicate(user_data, TRUE);
 }
 
+typedef struct dt_iop_gui_rename_module_t
+{
+  GtkWidget *floating_window;
+  dt_iop_module_t *module;
+} dt_iop_gui_rename_module_t;
+
+// http://stackoverflow.com/questions/4631388/transparent-floating-gtkentry
+static gboolean _rename_module_key_press(GtkWidget *entry, GdkEventKey *event, dt_iop_gui_rename_module_t *d)
+{
+  int ended = 0;
+
+  switch(event->keyval)
+  {
+    case GDK_KEY_Escape:
+      ended = 1;
+      break;
+    case GDK_KEY_Return:
+    case GDK_KEY_KP_Enter:
+    {
+      const gchar *name = gtk_entry_get_text(GTK_ENTRY(entry));
+      if(strcmp(d->module->multi_name, name) != 0)
+      {
+        g_strlcpy(d->module->multi_name, name, sizeof(d->module->multi_name) - 1);
+        d->module->multi_name[sizeof(d->module->multi_name) - 1] = 0;
```

Comment:
this is redundant as the g_strlcpy() doc makes it clear that dest is guaranteed to be null terminated.